code/controllers/subsystem/profiler.dm

```diff
@@ -0,0 +1,41 @@
+#define PROFILER_FILENAME "profiler.json"
+
+SUBSYSTEM_DEF(profiler)
+	name = "Profiler"
+	init_order = INIT_ORDER_PROFILER
+	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY
+	wait = 600
+
+/datum/controller/subsystem/profiler/Initialize()
+	if(CONFIG_GET(flag/auto_profile))
+		StartProfiling()
+	return ..()
+
+/datum/controller/subsystem/profiler/fire()
+	if(CONFIG_GET(flag/auto_profile))
+		DumpFile()
+
+/datum/controller/subsystem/profiler/Shutdown()
+	if(CONFIG_GET(flag/auto_profile))
+		DumpFile()
+	return ..()
+
+/datum/controller/subsystem/profiler/proc/StartProfiling()
+#if DM_BUILD < 1506
+	stack_trace("Auto profiling unsupported on this byond version")
+#else
+	world.Profile(PROFILE_START)
```

Comment:
It should be starting as early as possible if we have automatic profiling enabled, idealy as the first thing in /world/New so we can get profiling data from startup. I don't know if a regular config flag is good for this

Reply:
I guess i can start the profiler early if possible and just stop it during init if the flag is unset, not very clean solution though.